src/prted/pmix/pmix_server_gen.c

```diff
@@ -1203,23 +1206,68 @@ pmix_status_t pmix_server_iof_pull_fn(const pmix_proc_t procs[], size_t nprocs,
     return PMIX_ERR_NOT_SUPPORTED;
 }
 
-pmix_status_t pmix_server_stdin_fn(const pmix_proc_t *source,
-                                   const pmix_proc_t targets[], size_t ntargets,
-                                   const pmix_info_t directives[], size_t ndirs,
-                                   const pmix_byte_object_t *bo,
-                                   pmix_op_cbfunc_t cbfunc, void *cbdata)
+static void pmix_server_stdin_push(int sd, short args, void *cbdata)
 {
+    prrte_pmix_server_op_caddy_t *cd = (prrte_pmix_server_op_caddy_t*)cbdata;
     prrte_process_name_t dst;
+    pmix_byte_object_t *bo = (pmix_byte_object_t*)cd->server_object;
+    size_t n;
     int rc;
 
-    PRRTE_PMIX_CONVERT_PROCT(rc, &dst, &targets[0]);
-    prrte_iof.push_stdin(&dst, (uint8_t*)bo->bytes, bo->size);
+
+    for (n=0; n < cd->nprocs; n++) {
+        PRRTE_PMIX_CONVERT_PROCT(rc, &dst, &cd->procs[n]);
+
+        PRRTE_OUTPUT_VERBOSE((1, prrte_debug_output,
+                              "%s pmix_server_stdin_push to dest %s: size %zu",
+                              PRRTE_NAME_PRINT(PRRTE_PROC_MY_NAME),
+                              PRRTE_NAME_PRINT(&dst),
+                              bo->size));
+        prrte_iof.push_stdin(&dst, (uint8_t*)bo->bytes, bo->size);
+    }
 
 #if PMIX_NUMERIC_VERSION >= 0x00040000
     if (NULL == bo->bytes || 0 == bo->size) {
-        return PMIX_ERR_IOF_COMPLETE;
+        cd->cbfunc(PMIX_ERR_IOF_COMPLETE, cd->cbdata);
     }
+    else {
+        cd->cbfunc(PMIX_SUCCESS, cd->cbdata);
+    }
+#else
+    cd->cbfunc(PMIX_SUCCESS, cd->cbdata);
 #endif
 
-    return PMIX_OPERATION_SUCCEEDED;
+    PMIX_BYTE_OBJECT_FREE(bo, 1);
+    PMIX_PROC_FREE(cd->procs, cd->nprocs);
+    PRRTE_RELEASE(cd);
+}
+
+pmix_status_t pmix_server_stdin_fn(const pmix_proc_t *source,
+                                   const pmix_proc_t targets[], size_t ntargets,
+                                   const pmix_info_t directives[], size_t ndirs,
+                                   const pmix_byte_object_t *bo,
+                                   pmix_op_cbfunc_t cbfunc, void *cbdata)
+{
+    pmix_byte_object_t *bo_cpy = NULL;
+    pmix_proc_t *targets_cpy = NULL;
+    size_t n;
+
+    // We need to copy the object and the targets since we are shifting them
```

Comment:
It is okay to make these copies, but you shouldn't have to do so - the PMIx server library is _required_ to maintain those objects until the ```cbfunc``` is executed, which you don't do until after the call to ```push_stdin```. Only reason why you would need to is if ```push_stdin``` isn't copying the data - but a review of that code shows it indeed does copy it.

Still, no harm done and safer this way as you never know when someone might add another component that violates the "copy the data" code path.

Reply:
I see your point. Removing the copies does work, but the compiler is now warning about ignoring the `const` qualifier in `pmix_server_stdin_fn`. Let's keep it like this for now since it makes the compiler happier.